### PR TITLE
refactor: post mention preview using title or content

### PR DESCRIPTION
**Pull request by sshanzel**
Changes: 
- Improved generics on the notification context so we will only have to do it once.
- Check whether the mention was from the title or from the content, and display the appropriate preview.
- `doneTo` property whenever applicable as an optional property.

### CODE REVIEW

File: src/common/markdown.ts [hunk 1]
```diff
@@ -133,3 +133,12 @@ export const saveMentions = (
...
   return query.orIgnore().execute();
 };
+
+export const checkHasMention = (content: string, username: string) =>
+  content.split(' ').some((word) => {
+    if (word.charAt(0) !== '@') return false;
+
+    const mention = word.substring(1);
+
+    return mention === username;
```
Comment (idoshamun) [inline src/common/markdown.ts:143]: Wouldn't it be simpler to do
```suggestion
    return word === `@${username}`
```

File: src/notifications/types.ts [hunk 2]
```diff
@@ -55,6 +56,7 @@ export type NotificationSourceRequestContext = NotificationBaseContext & {
 
 export type NotificationDoneByContext = NotificationBaseContext & {
   doneBy: Reference<User>;
+  doneTo?: Reference<User>;
```
Comment (idoshamun) [inline src/notifications/types.ts:59]: What's the purpose of this field? Isn't that always the notification user?

File: src/notifications/generate.ts [hunk 3]
```diff
@@ -149,12 +150,17 @@ export const generateNotificationMap: Record<
...
     builder,
-    ctx: NotificationPostContext & NotificationDoneByContext,
+    ctx: NotificationPostContext<FreeformPost> & NotificationDoneByContext,
   ) =>
     builder
       .referencePost(ctx.post)
       .icon(NotificationIcon.Comment)
-      .description((ctx.post as WelcomePost).content, true)
+      .description(
+        checkHasMention(ctx.post.title, ctx.doneTo.username)
```
Comment (idoshamun) [inline src/notifications/generate.ts:159]: Let's add test to cover that, usually we don't check the description but here we have some logic to test

File: src/notifications/types.ts (hunk 2 above)
Comment (sshanzel) [reply src/notifications/types.ts:59]: Yes, but we only have the `userId` of the user to notify in this context but we need the username. Or do we have a pre-existing property for the relevant User object?

File: src/common/markdown.ts (hunk 1 above)
Comment (sshanzel) [reply src/common/markdown.ts:143]: Indeed, that is much simpler! 🤯

File: src/notifications/generate.ts (hunk 3 above)
Comment (sshanzel) [reply src/notifications/generate.ts:159]: I agree. I was looking for some examples and have found it now 🚢

**Review by idoshamun — APPROVED** [review]

**Pull request merged**